airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.ts

```diff
@@ -133,13 +181,29 @@ const calculateInitialCatalog = (
       const nodeWithId: SyncSchemaStream = { ...apiNode, id: id.toString() };
       const nodeStream = verifySourceDefinedProperties(verifySupportedSyncModes(nodeWithId), isNotCreateMode || false);
 
-      // if the stream is new since a refresh, we want to verify cursor and get optimal sync modes
-      const matches = newStreamDescriptors?.some(
-        (streamId) => streamId.name === nodeStream?.stream?.name && streamId.namespace === nodeStream.stream?.namespace
+      // if the stream is new since a refresh, verify cursor and get optimal sync modes
+      const isStreamNew = newStreamDescriptors?.some(
+        (streamIdFromDiff) =>
+          streamIdFromDiff.name === nodeStream?.stream?.name &&
+          streamIdFromDiff.namespace === nodeStream.stream?.namespace
       );
-      if (isNotCreateMode && !matches) {
-        return nodeStream;
+
+      // narrow down the breaking field changes from this connection to only those relevant to this stream
+      const breakingChangesByStream =
+        streamsWithBreakingFieldChanges && streamsWithBreakingFieldChanges.length > 0
+          ? streamsWithBreakingFieldChanges.filter((streamTransformFromDiff) => {
+              return (
+                streamTransformFromDiff.streamDescriptor.name === nodeStream?.stream?.name &&
+                streamTransformFromDiff.streamDescriptor.namespace === nodeStream.stream?.namespace
+              );
+            })
+          : [];
+
+      // if we're in edit or readonly mode and the stream is not new, check for breaking changes then return
+      if (isNotCreateMode && !isStreamNew) {
```

Comment:
I don't love that this check happens on `readonly` connections.  Presumably `readonly` connections wouldn't ever have a breaking change so should return immediately?  However looking at where `isNotCreateMode` is passed around, refactoring the code to pass the _actual_ mode is pretty hairy unfortunately.

It's not the only place we touch a `readonly` connection in this method (see `verifySourceDefinedProperties` being called on line 182 here) so maybe it's ok for now if not ideal.